lib/active_model/serializer/adapter/attributes.rb

```diff
@@ -74,6 +73,23 @@ def relevant_included_associations_for(serializer, nested_associations = {})
           end
         end
 
+        def add_relationship_for(parent, key, serializer, opts)
+          # mutates parent
+          apply_relationship_to(parent, key) do
+            if serializer.respond_to?(:each)
+              to_many_relationship_for(serializer, opts)
+            else
+              to_one_relationship_for(serializer, opts)
+            end
+          end
+
+          parent
+        end
```

Comment:
I'm good with this in principle, but I'd really like to separate the serializer.respond_to?(:each) flows using polymorphism rather than conditionals